Bag.hs

```diff
@@ -98,7 +98,8 @@ classifyFile f n classifyStrat = do
                         let parsedNewMessage = parseGrams wordBlackList n dlims newMessage
                         let newMessageVect = sparsifyVectSentence $ vectorizeSentence corpus parsedNewMessage
 
-                        let isSpam = classifyStrat vectSpams vectHams newMessageVect
+                        let isSpam = classSentence parsedSpams parsedHams parsedNewMessage
```

Comment:
If we do merge this in, can you not hardcode the classSentence function call here? It basically makes it so we can't use the cosine similarity strategy